homeassistant/components/switchbot/config_flow.py

```diff
@@ -466,5 +471,22 @@ async def async_step_init(
                     ): bool
                 }
             )
+        if (
+            CONF_SENSOR_TYPE in self.config_entry.data
+            and self.config_entry.data[CONF_SENSOR_TYPE] == SupportedModels.CURTAIN
+        ):
+            options.update(
+                {
+                    vol.Optional(
+                        CONF_CURTAIN_SPEED,
+                        default=self.config_entry.options.get(
+                            CONF_CURTAIN_SPEED, DEFAULT_CURTAIN_SPEED
+                        ),
+                    ): vol.All(
```

Comment:
Consider using a number selector here so we can configure the interface to be a box or a slider depending on what's most appropriate.

https://github.com/home-assistant/core/blob/60014b6530fd1b5d6d0849de7ad4050a89eed329/homeassistant/helpers/selector.py#L1121

I'm not sure what the frontend does by default here.